src/screens/Campaign.jsx

```diff
@@ -1,27 +1,62 @@
 import { useState, useEffect } from 'react';
-import { View, Text, ScrollView, Dimensions, StyleSheet } from 'react-native';
+import { View, Text, ScrollView, Dimensions, StyleSheet, Button } from 'react-native';
 import { StatusBar } from 'expo-status-bar';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import Header from '../features/campaigns/Header';
 import Footer from '../features/campaigns/Footer';
 import CharacterList from '../features/campaigns/CharacterList';
 import Notes from '../features/notes/Notes';
 import AddNote from '../features/notes/AddNote';
 import EncounterMenu from '../features/encounter/EncounterMenu';
+import QuestModal from '../features/quests/QuestModal.js'
+import {currentQuestUpdated, nextObjective, previousObjective} from '../features/campaigns/campaignSlice.js'
 
 export default function Campaign({ navigation }) {
 
+  const dispatch = useDispatch();
 
   const campaign = useSelector(state => (state.campaign))
   const [loading, setLoading] = useState(false)
+  const [isQuestModalVisible, setIsQuestModalVisible] = useState(false);
+
+  const toggleQuestModal = () => {
+    setIsQuestModalVisible(!isQuestModalVisible);
+  }
+
+  const setCurrentQuest = (questTitle) => {
+    dispatch(currentQuestUpdated(questTitle));
+  }
+
+  const campaignState = useSelector(state => state.campaign)
+
+  const currentObjectiveIndex = campaignState.currentObjectiveIndex
+  const currentQuestData = useSelector(state => state.campaign.quests.find(quest => quest.title === state.campaign.currentQuest));
+  const currentObjective = campaignState.quests.find(quest => quest.title === campaignState.currentQuest)
+
+  const handleNextObjective = () => {
+    console.log('next!')
+    dispatch(nextObjective());
+  }
+
+  const handlePreviousObjective = () => {
+    dispatch(previousObjective())
+  }
 
   if (loading) {
     return <StatusBar />
   }
 
+  const currentQuest = useSelector(state => state.campaign.currentQuest)
+
   return (
     <View style={styles.container}>
       <Header />
+      <Text style={styles.currentQuestTitle}>Current Quest: {currentQuest}</Text>
+      <Text>Current Objective: {currentQuestData && currentObjectiveIndex !== undefined ? currentQuestData.objectives[currentObjectiveIndex] : 'None'}</Text>
+      <Button title="Next Objective" onPress={handleNextObjective}></Button>
+      <Button title="Previous Objective" onPress={handlePreviousObjective}></Button>
+      <Button title="Show Quests" onPress={toggleQuestModal}></Button>
```

Comment:
Add an onPress handler to the Quest TouchableOpacity to either toggle this modal or navigate to a separate screen with buttons pertinent to quest management instead of creating a new button

Reply:
I added a TouchableOpacity modal for now, long-term we can add a journal page instead of (or in addition to) a quest modal for quest management